app/questionnaire/questionnaire_schema.py

```diff
@@ -215,19 +215,36 @@ def get_post_submission(self) -> ImmutableDict:
         schema: ImmutableDict = self.json.get("post_submission", ImmutableDict({}))
         return schema
 
+    def _check_list_exists(self, rules: Union[dict, Sequence], list_name: str) -> bool:
+        # old rules
+        if isinstance(rules, Sequence):
+            return any(
+                rule.get("list") == list_name and "list" in rule for rule in rules
+            )
+
+        # New rules
+        if isinstance(rules, dict) and rules.get("source") == "list":
+            return rules.get("identifier") == list_name
+
+        # for nested new rules
+        if isinstance(rules, dict):
+            operator = next(iter(rules))
+            operand = rules[operator]
+            return self._check_list_exists(operand[0], list_name)
```

Comment:
- I think this needs some more rework, I think it will be easier to use the logic used in the rules' evaluator on how it does recursion. This now assumes all dicts are rules, but we can have dicts that are not rules, such as date offsets. Similar to rules evaluator, we want to know it is a dict and a valid rule before doing recursion.
- Also that last line is only checking the first operand `operand[0]`, it needs to check all operands. Right now, if you swap the test operands around for the one where you expect to find a list it would fail because it isn't checking the 2nd operand which has the list rule.

Reply:
Updated this method to handle all nested cases (hopefully :-) )